.github/workflows/compare.yaml

```diff
@@ -0,0 +1,92 @@
+name: Compare Schemas for Changes
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+
+jobs:
+  compare:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+        with:
+          path: current-ref
+
+      - name: Fetch origin main
+        working-directory: current-ref
+        run: git fetch origin main
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: "pip"
+
+      - name: Install dependencies
+        working-directory: current-ref
+        run: |
+          python -m pip install --upgrade pip uv
+          uv pip install --system -r requirements.txt
+
+      - name: Set relative path to schema directory
+        run: |
+          echo "SCHEMA_DIR=python/lsst/sdm_schemas/schemas" >> $GITHUB_ENV
+
+      - name: Get list of changed YAML files
+        working-directory: current-ref
+        run: |
+          CHANGED_FILES=$(git diff --name-only origin/main..HEAD -- ${{ env.SCHEMA_DIR }} | grep '\.yaml$')
```

Comment:
This logic needs to be changed.  To avoid future surprises, it must compare the "current ref" against the base of the branch in which it appears.  This only really makes sense for PRs, because what we're trying to do is to ask the question "if this PR is merged, will it produce a schema change?".  It's true that 90%+ of the time it's the same as comparing to the tip of `main`, but not always, and we need to get this right up front.